src/plugins/intel_cpu/src/transformations/snippets/x64/pass/fuse_brgemm_cpu_postops.cpp

```diff
@@ -0,0 +1,434 @@
+// Copyright (C) 2025 Intel Corporation
+// SPDX-License-Identifier: Apache-2.0
+//
+
+#include "fuse_brgemm_cpu_postops.hpp"
+
+#include "memory_desc/dnnl_blocked_memory_desc.h"
+#include "openvino/pass/pattern/matcher.hpp"
+#include "openvino/pass/pattern/op/optional.hpp"
+#include "openvino/pass/pattern/op/or.hpp"
+#include "openvino/pass/pattern/op/wrap_type.hpp"
+#include "snippets/itt.hpp"
+#include "snippets/op/convert_saturation.hpp"
+#include "snippets/op/rank_normalization.hpp"
+#include "snippets/op/scalar.hpp"
+#include "transformations/snippets/x64/op/brgemm_cpu.hpp"
+#include "transformations/snippets/x64/op/brgemm_utils.hpp"
+
+namespace ov::intel_cpu {
+
+using namespace dnnl::impl;
+using namespace snippets::op;
+using namespace snippets::lowered;
+using namespace ov::pass::pattern;
+using PortDescriptorUtils = snippets::lowered::PortDescriptorUtils;
+
+namespace {
+std::shared_ptr<BrgemmCPU> clone_with_new_params(
+    const std::shared_ptr<const BrgemmCPU>& brgemm,
+    const BrgemmCPU::PostopsConfig& postops,
+    const ov::OutputVector& new_inputs,
+    const std::vector<ov::snippets::modifier::MemoryAccess::PortDescriptor>& new_in_descs) {
+    auto new_brgemm =
+        std::make_shared<BrgemmCPU>(new_inputs,
+                                    brgemm->get_type(),
+                                    new_in_descs,
+                                    brgemm->get_output_port_descriptor(0),
+                                    PortDescriptorUtils::get_port_descriptor_ptr(brgemm->input(0))->get_layout(),
+                                    PortDescriptorUtils::get_port_descriptor_ptr(brgemm->input(1))->get_layout(),
+                                    PortDescriptorUtils::get_port_descriptor_ptr(brgemm->output(0))->get_layout(),
+                                    postops);
+
+    // PortDescriptors are copied manually since it is not copyable attribute
+    for (size_t i = 0; i < brgemm->get_input_size(); ++i) {
+        const auto in_desc = PortDescriptorUtils::get_port_descriptor_ptr(brgemm->input(i));
+        PortDescriptorUtils::set_port_descriptor_ptr(new_brgemm->input(i), in_desc);
+    }
+    const auto out_desc = PortDescriptorUtils::get_port_descriptor_ptr(brgemm->output(0));
+    PortDescriptorUtils::set_port_descriptor_ptr(new_brgemm->output(0), out_desc);
+    return new_brgemm;
+}
+
+auto brgemm_predicate = [](const Output<Node>& output) {
+    const auto brgemm = output.get_node_shared_ptr();
+    // Note: postops are not supported in case of blocking enabled,
+    // so f32 precision is not included in supported list
+    // Ticket: 165567
+    static const ov::element::TypeVector supported_in_precisions{ov::element::bf16, ov::element::i8, ov::element::u8};
```

Comment:
BTW We also support `fp16`.

What's about to check `brgemm_type` instead of precision - it should not be `stand alone` ?

Also can we add additional check in blocking pass that if `BrgemmCPU` node has postops - blocking loops are prohibited? It would be safer

Reply:
I recalled that `fp32` `MatMul` with `transposed_b=true` is `Brgemm with_repacking` too🤔 

What's about to create static method `BrgemmCPUBlocking::supports_brgemm_kn_blocking()` which can be called in `BrgemmCPUBlocking` pass and here in `brgemm_predicate`? To avoid code dependencies.

What do you think?

Reply:
`BrgemmCPUBlocking::is_kn_blocking_supported` is introduced: it is used in `brgemm_predicate` and `BrgemmCPUBlocking::get_blocking_params`. In `get_blocking_params`, I also added an assert that in case of blocking is supported, brgemm mustn't have any postops.